specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2024-01-01-preview/Recommendations.json

```diff
@@ -208,79 +233,52 @@
           "type": "string"
         },
-        "recommendationTypeTitle": {
-          "description": "Title of the recommendation type.",
-          "type": "string"
-        },
-        "recommendationTypeId": {
-          "description": "Id of the recommendation type.",
-          "type": "string"
-        },
-        "category": {
-          "description": "Category of the recommendation.",
-          "$ref": "#/definitions/RecommendationCategory"
+        "creationTimeUtc": {
+          "description": "The time stamp (UTC) when the recommendation was created.",
+          "type": "string",
+          "format": "date-time"
         },
-        "context": {
-          "description": "Context of the recommendation.",
-          "$ref": "#/definitions/RecommendationContext"
+        "lastEvaluatedTimeUtc": {
+          "description": "The time stamp (UTC) when the recommendation was last evaluated.",
+          "type": "string",
+          "format": "date-time"
         },
-        "workspaceId": {
-          "description": "Id of the workspace this recommendation refers to.",
+        "lastModifiedTimeUtc": {
+          "description": "The time stamp (UTC) when the recommendation was last modified.",
           "type": "string",
-          "format": "uuid"
+          "format": "date-time"
         },
-        "actions": {
-          "description": "List of actions to take for this recommendation.",
+        "suggestions": {
+          "description": "List of suggestions to take for this recommendation.",
           "type": "array",
           "items": {
-            "$ref": "#/definitions/RecommendedAction"
+            "$ref": "#/definitions/RecommendedSuggestion"
           },
           "x-ms-identifiers": []
         },
-        "state": {
-          "description": "State of the recommendation.",
-          "$ref": "#/definitions/RecommendationState"
-        },
-        "priority": {
-          "description": "Priority of the recommendation.",
-          "$ref": "#/definitions/RecommendationPriority"
-        },
-        "lastEvaluatedTimeUtc": {
-          "description": "The time stamp (UTC) when the recommendation was last evaluated.",
-          "type": "string",
-          "format": "date-time"
-        },
-        "hideUntilTimeUtc": {
-          "description": "The time stamp (UTC) when the recommendation should be displayed again.",
+        "resourceId": {
+          "description": "Id of the resource this recommendation refers to.",
           "type": "string",
-          "format": "date-time",
           "x-nullable": true
         },
-        "displayUntilTimeUtc": {
-          "description": "The timestamp (UTC) after which the recommendation should not be displayed anymore.",
-          "type": "string",
-          "format": "date-time",
+        "additionalProperties": {
+          "description": "Collection of additional properties for the recommendation.",
+          "type": "object",
+          "additionalProperties": {
```

Comment:
>additionalProperties [](http://example.com/codeflow?start=11&length=20)

Use of additionalProperties is not allowed in Azure. Please provide a justification for why this is required and why it cannot be a concrete type that is not a dictionary

Reply:
First, the `additionalProperties` field was already available on the previous versions of this API.
Secondly, this API is a generic API aimed at providing a general solution for multiple partners in our group.
New recommendations will be added at a fast pace, and supporting and maintaining a new API version for each type of recommendation will be incredibly challenging.
Also, the content pushed there differs for each user environment - (i.e. each user might have different content inside this field depending on their configurations and environment).